src/witan/workspace/protocols.clj

```diff
@@ -2,3 +2,18 @@
 
 (defprotocol SendMessage
   (send-message! [this topic message]))
+
+;;;;;;;;;
+
+(defprotocol Database
+  (drop-table!
+    [this table])
+  (create-table!
+    [this table columns])
+  (insert!
+    [this table row]
+    [this table row args])
+  (select*
```

Comment:
Not super convinced that this presents a huge plus over hayt's dsl, and we only have the one Database atm.  What was your motivation?